business-networks-workflows/src/main/kotlin/net/corda/bn/flows/ApproveMembershipAttributeChangeFlow.kt

```diff
@@ -0,0 +1,96 @@
+package net.corda.bn.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.bn.contracts.ChangeRequestContract
+import net.corda.bn.states.ChangeRequestState
+import net.corda.bn.states.ChangeRequestStatus
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.flows.FlowException
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.FlowSession
+import net.corda.core.identity.Party
+import net.corda.core.internal.uncheckedCast
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+
+/**
+ * This flow is initiated by an authorised member who wants to approve a [MembershipModificationRequest] to change
+ * a members attributes.
+ * Modifies an existing [ChangeRequestState]'s status to APPROVED on initiator's and authorised member's ledgers and
+ * executes the changes.
+ *
+ * @property requestId The ID of the request which needs to be accepted.
+ * @property notary Identity of the notary to be used for transactions notarisation. If not specified, first one from the whitelist will be used.
+ * */
+@InitiatingFlow
+@StartableByRPC
+class ApproveMembershipAttributeChangeFlow(
+        private val requestId: UniqueIdentifier,
+        private val notary: Party? = null
+) : MembershipManagementFlow<SignedTransaction>() {
+
+    @Suspendable
+    override fun call(): SignedTransaction {
+        val bnService = serviceHub.cordaService(BNService::class.java)
```

Comment:
Totally unrelated proposal, but what do you think about moving this property in the parent class? all flows seem to create it over and over again.

Reply:
As discussed offline, the BNService cannot be retrieved before the Flow is instantiated